### PR TITLE
FPGA APIs initial setup

### PR DESCRIPTION
**Pull request by amehra-ni**
### What does this Pull Request accomplish?
Add nifpga service to grpc-device server. This PR contains metadata for a few functions that are not marked as 'codegen_method': 'no'. The functions are **Abort, Close,  Download, Open, Reset, CommitFifoConfiguration, StartFifo, StopFifo, UnreserveFifo, ReleaseFifoElements,and Run**.

### Why should this Pull Request be merged?
- Registers nifpga service with grpc-device server with NextRelease code readiness.
- Adds an initial set of FPGA APIs to the repo to get started with parallel development.
- The .h file and NiFpga.dll have different function names( NiFpga_Open vs NiFpgaDll_Open). To call it correctly, I've added a new parameter(c_dll_function_prefix) in config.py and updated the library codegen code to correctly load NiFpgadll function names.
- ConvertApiErrorStatusForNiFpga_Session: In other services, here we usually call a function that returns a description of the error code, but in FPGA I didn't see any function that will do that. We would probably need to check the status value we got and return the description by some parsing logic(there are status codes in the FPGA header and it has some summary), but that would be in a follow-up, hence a hard-coded string for now.

### What testing has been done?
- No compile or codegen issues.
- Manually verified server handlers for all APIs and compared them with documentation for the APIs mentioned.
- Manually tested a few APIs with client code which calls the functions correctly.

### CODE REVIEW

**Review by astarche — APPROVED** [review]

**Review by reckenro — APPROVED** [review]

**Review by maxxboehme — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -91,7 +91,7 @@ set(metadata_dir ${CMAKE_SOURCE_DIR}/source/codegen/metadata)
 set(service_output_dir ${CMAKE_SOURCE_DIR}/generated)
 set(codegen_dir ${CMAKE_SOURCE_DIR}/source/codegen)
 set(custom_dir ${CMAKE_SOURCE_DIR}/source/custom)
-set(nidrivers "nidaqmx" "nidcpower" "nidigitalpattern" "nidmm" "nifake" "nifake_extension" "nifake_non_ivi" "nifgen" "nimxlcterminaladaptor_restricted" "nirfmxbluetooth" "nirfmxcdma2k" "nirfmxgsm" "nirfmxinstr" "nirfmxinstr_restricted" "nirfmxlte" "nirfmxnr" "nirfmxspecan" "nirfmxspecan_restricted" "nirfmxtdscdma" "nirfmxwcdma" "nirfmxdemod" "nirfmxwlan" "nirfmxwlan_restricted" "nirfsa" "nirfsg" "nirfsg_restricted" "niscope" "niscope_restricted" "niswitch" "nisync" "nitclk" "nixnet" "nixnetsocket" "visa")
+set(nidrivers "nidaqmx" "nidcpower" "nidigitalpattern" "nidmm" "nifake" "nifake_extension" "nifake_non_ivi" "nifgen" "nimxlcterminaladaptor_restricted" "nirfmxbluetooth" "nirfmxcdma2k" "nirfmxgsm" "nirfmxinstr" "nirfmxinstr_restricted" "nirfmxlte" "nirfmxnr" "nirfmxspecan" "nirfmxspecan_restricted" "nirfmxtdscdma" "nirfmxwcdma" "nirfmxdemod" "nirfmxwlan" "nirfmxwlan_restricted" "nirfsa" "nirfsg" "nirfsg_restricted" "niscope" "niscope_restricted" "niswitch" "nisync" "nitclk" "nixnet" "nixnetsocket" "visa" "nifpga")
```
Comment (doshirohan) [inline CMakeLists.txt:94]: Add `nifpga` before `nifgen` to keep it in sorted order.

File: source/codegen/metadata/nifpga/config.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    'c_header': 'NiFpga.h',
+    'c_function_prefix': 'NiFpga_',
+    'c_dll_function_prefix': 'NiFpgaDll_',
+    'service_class_prefix': 'NiFpga',
+    'java_package': 'com.ni.grpc.nifpga',
+    'csharp_namespace': 'NationalInstruments.Grpc.NiFpga',
+    'namespace_component': 'nifpga',
+    'init_function': 'Open',
+    'close_function': 'Close',
+    'code_readiness': 'NextRelease',
```
Comment (doshirohan) [inline source/codegen/metadata/nifpga/config.py:13]: Do have a task under your feature to remove this feature toggle when you are near done.

**Review by doshirohan — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (amehra-ni) [reply CMakeLists.txt]: updated

File: source/codegen/metadata/nifpga/config.py (hunk 2 above)
Comment (amehra-ni) [reply source/codegen/metadata/nifpga/config.py:13]: will do that.

File: source/codegen/metadata/nifpga/config.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+config = {
+    'api_version': '24.3.0',
+    'c_header': 'NiFpga.h',
+    'c_function_prefix': 'NiFpga_',
```
[the remark this answers is not in the document]
Comment (amehra-ni) [reply source/codegen/metadata/nifpga/config.py:5]: Will be closing out the PR.
As code generation would not be affected I can create a separate task if any changes are required.

**Pull request merged**